### PR TITLE
Add a GitHub Actions CI system

### PR DESCRIPTION
**Pull request by TrentHouliston**

### CODE REVIEW

**Comment by Bidski** [comment]
You and your github actions @TrentHouliston !!

**Comment by JosephusPaye** [comment]
> You and your github actions @TrentHouliston !!

I know right 😅

GH Actions here should be fine though, since this repo might one day be transferred to the RoboCup organization, and we wouldn't want them to have to setup their own Buildkite agents.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Validate the Dashboard code style and formatting are correct according to eslint and prettier
+  dashbaord-format:
+    name: Dashboard Formatting
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Run on the Node 12 container
+    container: node:12
```
Comment (JosephusPaye) [inline .github/workflows/main.yaml]: We can probably bump to Node 14, the latest LTS.

**Review by JosephusPaye — DISMISSED** [review]
How can I say no to free work 😛? Just one change.

**Comment by JosephusPaye** [comment]
Also, does the caching `yarn` actually work? I remember we had issues setting it up on Horus.

**Comment by TrentHouliston** [comment]
That environment variable doesn't actually do caching, it just sets it to a folder so it works properly. Since it's in the cloud you can probably do the cache in a future PR

https://github.com/actions/cache/blob/master/examples.md#node---yarn

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      - name: Run eslint
+        run: yarn eslint
+
+  # Build the Dashboard code
+  build-dashboard:
+    name: Build Dashboard
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Run on the Node 12 container
```
Comment (Bidski) [inline .github/workflows/main.yaml]:
```suggestion
    # Run on the Node 14 container
```

File: .github/workflows/main.yaml (hunk 1 above)
Comment (Bidski) [inline .github/workflows/main.yaml]:
```suggestion
    # Run on the Node 14 container
    container: node:14
```

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        run: yarn build
+
+  # Build the Dashboard Storybook code
+  build-dashboard-storybook:
+    name: Build Dashboard Storybook
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Run on the Node 12 container
+    container: node:12
```
Comment (Bidski) [inline .github/workflows/main.yaml]:
```suggestion
    # Run on the Node 14 container
    container: node:14
```

File: .github/workflows/main.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        run: yarn storybook:build
+
+  # Test the Dashboard code
+  test-dashboard:
+    name: Test Dashboard
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    # Run on the Node 12 container
+    container: node:12
```
Comment (Bidski) [inline .github/workflows/main.yaml]:
```suggestion
    # Run on the Node 14 container
    container: node:14
```

**Review by Bidski — CHANGES_REQUESTED** [review]

**Review by Bidski — APPROVED** [review]
Looks ok, now you just need to figure out why the checks are failing!

**Comment by JosephusPaye** [comment]
Tests are failing due to the missing proto messages, which was fixed in #3. Merging master should fix this.

**Review by JosephusPaye — APPROVED** [review]
LGTM

**Pull request merged**